.github/workflows/mpt-witness-generator.yml

```diff
@@ -0,0 +1,47 @@
+name: MPT Witness Generator Checks
+
+on:
+  merge_group:
+  pull_request:
+    types: [synchronize, opened, reopened, ready_for_review]
+  push:
+    branches:
+      - main
+
+jobs:
+  skip_check:
+    runs-on: ubuntu-latest
+    outputs:
+      should_skip: ${{ steps.skip_check.outputs.should_skip }}
+    steps:
+      - id: skip_check
+        uses: fkirc/skip-duplicate-actions@v5
+        with:
+          cancel_others: 'true'
+          concurrent_skipping: 'same_content_newer'
+          paths: '["mpt-witness-generator/**"]'
+
+
+  build:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+
+    - name: Set up Go
+      uses: actions/setup-go@v4
+      with:
+        go-version: '1.20'
+
+    - name: Format
+      working-directory: ./mpt-witness-generator
+      run: go fmt ./...
```

Comment:
I think this will not report formatting issues.  Instead it will try to format all the code and return success (no matter if there were changes or not).

Maybe we can use this action, which fails when the code is not formatted https://github.com/Jerome1337/gofmt-action

Reply:
Nice catch! Fixed in [b84da75](https://github.com/privacy-scaling-explorations/zkevm-circuits/pull/1591/commits/b84da758cc3d38aed68d8db47c074419518cfe0f)